src/main/java/com/iheartradio/m3u8/Encoding.java

```diff
@@ -9,4 +12,10 @@
     private Encoding(String value) {
         this.value = value;
     }
+    
+    public static Encoding byEncoding(String encodingName) throws NoSuchElementException {
+        return Arrays.stream(Encoding.values()).filter(e -> {
```

Comment:
Please remove the lambda syntax. This library needs to support Android which means we are limited to java 7 sans try-with-resources.